src/LowestCommonAncestor.php

```
@@ -2,35 +2,55 @@
-    $vertices_local = $bfs_local->setDirection($direction)->getVertices()->getMap();
-    // Use BFS algorithm to get all vertices starting with $remote to the root.
-    $bfs_remote = new BreadthFirst($remote);
-    $vertices_remote = $bfs_remote->setDirection($direction)->getVertices()->getMap();
-    // Intersect the $vertices_local and $vertices_remote to get all common
-    // ancestors for $local and $remote.
-    $vertices_intersection = array_intersect(array_keys($vertices_local), array_keys($vertices_remote));
-    // If the $vertices_intersection array is not empty then first value should
-    // be our solution - the LCA.
-    $lca_id = reset($vertices_intersection);
-    if ($lca_id !== NULL && $lca = $this->graph->getVertex($lca_id)) {
-      return $lca;
+        try
+        {
+            // Use BFS algorithm to get all vertices starting with $local to the root.
+            $bfs_local = new BreadthFirst($local);
+            $vertices_local = $bfs_local->setDirection($direction)->getVertices();
+            // Use BFS algorithm to get all vertices starting with $remote to the root.
+            $bfs_remote = new BreadthFirst($remote);
+            $vertices_remote = $bfs_remote->setDirection($direction)->getVertices();
+            // Intersect the $vertices_local and $vertices_remote to get all common
+            // ancestors for $local and $remote. If the $vertices_intersection array is
+            // not empty then first value should be our solution - the LCA.
+            $vertices_intersection = $vertices_local
+                ->getVerticesIntersection($vertices_remote)
+                ->getMap();
+            if($vertices_intersection!= NULL) {
+                return reset($vertices_intersection) ?: FALSE;
+            }
+            else{
+                throw new UnderflowException("No common ancestor found");
+            }
+        }
+        catch (Exception $e)
+        {
+            echo $e->getMessage();
```

It's never a good idea to `echo` results in a method because it can cause unwanted buffer output. So it's better to just throw the exception. It's better to leave it up to the caller to handle the exception.